uberfire-commons-editor/uberfire-commons-editor-backend/src/test/java/org/uberfire/ext/editor/commons/backend/service/utils/PathNameUtilsTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.ext.editor.commons.backend.service.utils;
+
+import java.io.IOException;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.uberfire.backend.server.util.Paths;
+import org.uberfire.backend.vfs.PathFactory;
+import org.uberfire.java.nio.file.Path;
+import org.uberfire.mocks.FileSystemTestingUtils;
+
+import static org.junit.Assert.*;
+
+public class PathNameUtilsTest {
```

Comment:
It might have been an idea to write this as a "parameterized" test (e.g. http://www.java2s.com/Tutorials/Java/JUnit/0140__JUnit_Parameterized_Test.htm) but nothing wrong with your approach.